### PR TITLE
File upload failure checking

### LINKED ISSUE #2712: Uploading a video fails.
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Expected Result:
1. A error message should be thrown if video uploading fails.
2. User should not be able to select Mkv videos as they are not supported from the backend.

## Actual Result:
1. Message is always in loading state.
Error thrown
```
Uncaught (in promise) TypeError: can't access property "sequenceNumber", reportAction is undefined
```

## Action Performed:
1. Open Expensify.cash on web browser.
2. Open any report.
3. try sending an MKV video.

## Workaround:
Covert the video to mp4 and then upload.

## Platform:
Where is this issue occurring?

 Web - :ballot_box_with_check: 
 iOS
 Android
 Desktop App :ballot_box_with_check: 
 Mobile Web :ballot_box_with_check: 

**Version Number:** V.1.0.38
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

Fixes #2712

### Tests |  QA Steps
1. Open any conversation on E.cash.
2. Now try to send a video that is not supported. For example any mkv file.
3. you should see an error message that the file is not supported and the sending message should be removed from the List.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/24370807/120571168-32d50a00-c437-11eb-9916-a67414665a41.mp4

#### Mobile Web

https://user-images.githubusercontent.com/24370807/120585102-c6fe9b80-c44e-11eb-8db2-e892ae5f8556.mp4

#### Android

https://user-images.githubusercontent.com/24370807/120584110-fc09ee80-c44c-11eb-9371-9abe5416e025.mp4

### CODE REVIEW

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -1040,7 +1041,17 @@ function addAction(reportID, text, file) {
         // the same way report actions can.
         persist: !isAttachment,
     })
-        .then(({reportAction}) => updateReportWithNewAction(reportID, reportAction));
+        .then((response) => {
+            if (response.jsonCode === 408) {
+                Growl.show('Upload Failed. File is not supported.', CONST.GROWL.ERROR);
```
Comment (parasharrajat) [inline src/libs/actions/Report.js:1046]: @iwiznia We should show show the localized string here. So I am pinging you to ask what is the best approach, there is not component involved and thus HOC can't be used.
What I think is that we can subscribe to the `prefferedLocale` using Onyx.Connect and then call the translate directly.

Comment (iwiznia) [reply src/libs/actions/Report.js:1046]: Yeah, that's the way to go.

**Comment by parasharrajat** [comment]
Ready for review. Thanks.

**Review by ctkochan22 — APPROVED** [review]

**Comment by parasharrajat** [comment]
@ctkochan22 :shipit:

**Comment by parasharrajat** [comment]
Is there anything left that is preventing it from being merged?

**Comment by ctkochan22** [comment]
Was waiting on @iwiznia but small change. Looks fine.

**Comment by isagoico** [comment]
@parasharrajat @ctkochan22 Anything that needs to be QAd here?

**Comment by parasharrajat** [comment]
Oh yeah @isagoico Updating steps.

**Comment by parasharrajat** [comment]
Done

**Comment by isagoico** [comment]
We were unable to test this PR in iOS because of this issue https://github.com/Expensify/Expensify.cash/issues/3543. All other environments (Android, Desktop and Web) were a pass.

**Pull request merged**